spec/electric_slide/call_queue_spec.rb

```diff
@@ -76,19 +76,24 @@
         allow(agent_call).to receive(:dial)
         queue.add_agent agent
         queue.enqueue queued_call
-        queued_call << Punchblock::Event::Joined.new(timestamp: connected_time)
-        agent_call << Punchblock::Event::Joined.new(timestamp: connected_time)
       end
 
       it "sets the agent's `call` attribute" do
         expect(agent.call).to be agent_call
       end
 
-      it "records the connection time in the :electric_slide_connected_at call variable on the queued call" do
-        expect(queued_call[:electric_slide_connected_at]).to eq(connected_time)
+      it 'records the agent in the `:agent` call variable on the queued call' do
+        expect(queued_call[:agent]).to eq(agent)
       end
 
       context 'when the call ends' do
+        before do
+          # "drop" the queued call also so that it doesn't get re-enqueued, and
+          # isn't re-connected to the agent's call, thus changing the agent
+          # call's state before it is checked in the examples
+          allow(queued_call).to receive(:active?).and_return(false)
```

Comment:
Could you try writing tests which don't mock this? I don't think you can rely on this because each call gets hung up independently.